laravel_versioned_docs/version-3.x/reference/30-plugin-api.md

```diff
@@ -56,14 +56,15 @@ This is the method that is called to process a route. Parameters:
 - `$routeRules`, the rules passed in the `apply` section of the Scribe config for this route.
 
 This method may return `null` or an empty array if it has no data to add. Otherwise, it should return an array with the relevant information, which varies depending on the type of strategy/stage of route processing:
-- For `metadata`, a map (key => value) of metadata attributes.
+- For `metadata`, a map (key => value) of metadata attributes. Modify the `$endpointData->metadata->custom` array directly if you want to set custom metadata, then return null.
```

Comment:
(The return value does not matter.)

```suggestion
- For `metadata`, a map (key => value) of metadata attributes (as shown in the example below). If you'd like to set a custom attribute so you can access it later, you can add items to the `$endpointData->metadata->custom` array directly.
- ```

Reply:
I see now that returning does matter because of the annotation, but I think mentioning "return null" is unnecessary baggage. The line above literally says "This method may return `null` or an empty array if it has no data to add".